src/Unused/Parser/Internal.hs

```diff
@@ -0,0 +1,45 @@
+module Unused.Parser.Internal
+    ( parseTermMatches
+    , parseTermMatch
+    , parse
+    , ParseError
+    ) where
+
+import Control.Monad (void)
+import Text.Parsec
+import Text.Parsec.String (Parser)
+import Unused.Types (TermMatch(..))
+
+parseTermMatches :: Parser [TermMatch]
+parseTermMatches = many1 parseTermMatch <* eof
+
+parseTermMatch :: Parser TermMatch
+parseTermMatch = do
+    colonSep
+    path <- pathParser
+    colonSep
+    occurrences <- occurrenceParser
+    colonSep
+    term <- termParser
+    void eol
+
+    return $ TermMatch term path $ toInt occurrences
+  where
+    toInt i = read i :: Int
```

Comment:
@sidraval since `occurrenceParser` is currently parsing with `many1 digit`, would it be a parser error instead?

After typing this out, though, the fact that there's ambiguity is cause for concern.

Unrelated - I think it probably makes sense to make it `occurrenceParser :: Parser Int` so `parseTermMatch` doesn't need to care how it gets the `Int` value.